src/posix/client.cpp

```diff
@@ -195,8 +202,11 @@ int main(int argc, char *argv[])
     while (!isFinished)
     {
         fd_set readFdSet;
-        char   buffer[OPENTHREAD_CONFIG_DIAG_CMD_LINE_BUFFER_SIZE];
-        int    maxFd = sSessionFd;
+        char   lineBuffer[kLineBufferSize];
+        char   buffer[kLineBufferSize];
+        size_t lineBufferWritePos = 0;
+        bool   isBeginOfLine      = true;
```

Comment:
lineBuffer, lineBufferWritePos, isBeginOfLine should be moved out of this scope. We should not re-initialize these values in each loop. 
@bukepo